### PR TITLE
KAA-1347: Failed test 12 on MacOSX

### PR DESCRIPTION
**Pull request by vasner**
Add platform independent sockaddr test
Reviewers: @rasendubi and @forGGe

### CODE REVIEW

File: client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_bootstrap.c [hunk 1]
```diff
@@ -730,16 +730,23 @@ kaa_error_t kaa_platform_protocol_alloc_serialize_client_sync(kaa_platform_proto
 
 ext_tcp_socket_state_t ext_tcp_utils_tcp_socket_check(kaa_fd_t fd, const kaa_sockaddr_t *destination, kaa_socklen_t destination_size)
 {
+    int result_memcmp;
```
Comment (rasendubi) [inline client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_bootstrap.c]: Reduce the scope of the variable. (see next comment)

File: client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_bootstrap.c [hunk 2]
```diff
@@ -730,16 +730,23 @@ kaa_error_t kaa_platform_protocol_alloc_serialize_client_sync(kaa_platform_proto
...
+
+    /* Only to avoid compiler's warning */
+    (void)destination_size;
 
     if (fd == ACCESS_POINT_SOCKET_FD) {
-        if(memcmp(destination, &DESTINATION_SOCKADDR,destination_size) == 0) {
+        result_memcmp = memcmp(destination->sa_data, DESTINATION_SOCKADDR.sa_data,
```
Comment (rasendubi) [inline client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_bootstrap.c]:
``` c
int result_memcmp = memcmp(destination->sa_data, DESTINATION_SOCKADDR.sa_data,
```

File: client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_bootstrap.c (hunk 1 above)
Comment (vasner) [inline client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_bootstrap.c]: How about MISRA compliance?

**Comment by Slash32** [comment]
@kaa-leeroy test this please

File: client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_bootstrap.c (hunk 1 above)
Comment (rasendubi) [inline client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_bootstrap.c]: The only rule that affects variable scope in MISRA-2012 is `Rule 8.9 An object should be defined at block scope if its identifier only appears in a single function`.  It also has the following comment:

> Within a function, whether objects are defined at the outermost or innermost block is largely a matter of style.

Comment (vasner) [inline client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_bootstrap.c]: Never mind (I was thinking about 5.2 for future). Fixed

File: client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_bootstrap.c [hunk 3]
```diff
@@ -730,16 +730,21 @@ kaa_error_t kaa_platform_protocol_alloc_serialize_client_sync(kaa_platform_proto
 
 ext_tcp_socket_state_t ext_tcp_utils_tcp_socket_check(kaa_fd_t fd, const kaa_sockaddr_t *destination, kaa_socklen_t destination_size)
 {
+    /* Only to avoid compiler's warning */
```
Comment (rasendubi) [inline client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_bootstrap.c:733]: You can avoid such comments in the future. Such usages of cast to void are common.

**Comment by rasendubi** [comment]
LGTM

File: client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_bootstrap.c (hunk 3 above)
Comment (vasner) [inline client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_bootstrap.c:733]: It came from India team times and I still adding it :)

**Comment by Slash32** [comment]
LGTM

**Pull request merged**